cmd/kubeadm/app/util/config/cluster.go

```diff
@@ -68,8 +69,13 @@ func getInitConfigurationFromCluster(kubeconfigDir string, client clientset.Inte
 		return nil, errors.Wrap(err, "failed to get config map")
 	}
 
-	// InitConfiguration is composed with data from different places
+	// Take an empty versioned InitConfiguration, statically default it and convert it to the internal type
+	versionedInitcfg := &kubeadmapiv1.InitConfiguration{}
+	kubeadmscheme.Scheme.Default(versionedInitcfg)
```

Comment:
FYI, this calls:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/v1beta3/defaults.go#L74-L79

the problem in the linked issue is only caused by the lack of calling:
- SetDefaults_NodeRegistration()

but also calling:
- SetDefaults_BootstrapTokens(obj)
- SetDefaults_APIEndpoint(&obj.LocalAPIEndpoint)

is fine because:
- tokens receive sane defaults, like tiimeouts
- apiendpoint receives the default port, but is later overriden by the dynamic defaults in this getInitConfigurationFromCluster() function.